libs/@blockprotocol/graph/src/types/subgraph/edges/variants/ontology.ts

```diff
@@ -0,0 +1,205 @@
+    rightEndpoint: OntologyTypeVertexId;
+  }
+>;
+
+export const isValuesConstrainedByEdge = (
+  outwardEdge: OutwardEdge,
+): outwardEdge is ValuesConstrainedByEdge => {
+  return outwardEdge.kind === "CONSTRAINS_VALUES_ON" && outwardEdge.reversed;
+};
+
+export type ConstrainsPropertiesOnEdge = Subtype<
+  GenericOutwardEdge,
+  {
+    reversed: false;
+    kind: "CONSTRAINS_PROPERTIES_ON";
+    rightEndpoint: OntologyTypeVertexId;
+  }
+>;
+
+export const isConstrainsPropertiesOnEdge = (
+  outwardEdge: OutwardEdge,
+): outwardEdge is ConstrainsPropertiesOnEdge => {
+  return (
+    outwardEdge.kind === "CONSTRAINS_PROPERTIES_ON" && !outwardEdge.reversed
+  );
+};
+
+export type PropertiesConstrainedByEdge = Subtype<
+  GenericOutwardEdge,
+  {
+    reversed: true;
+    kind: "CONSTRAINS_PROPERTIES_ON";
+    rightEndpoint: OntologyTypeVertexId;
+  }
+>;
+
+export const isPropertiesConstrainedByEdge = (
+  outwardEdge: OutwardEdge,
+): outwardEdge is PropertiesConstrainedByEdge => {
+  return (
+    outwardEdge.kind === "CONSTRAINS_PROPERTIES_ON" && outwardEdge.reversed
+  );
+};
+
+export type ConstrainsLinksOnEdge = Subtype<
+  GenericOutwardEdge,
+  {
+    reversed: false;
+    kind: "CONSTRAINS_LINKS_ON";
+    rightEndpoint: OntologyTypeVertexId;
+  }
+>;
+
+export const isConstrainsLinksOnEdge = (
+  outwardEdge: OutwardEdge,
+): outwardEdge is ConstrainsLinksOnEdge => {
+  return outwardEdge.kind === "CONSTRAINS_LINKS_ON" && !outwardEdge.reversed;
+};
+
+export type LinksConstrainedByEdge = Subtype<
+  GenericOutwardEdge,
+  {
+    reversed: true;
+    kind: "CONSTRAINS_LINKS_ON";
+    rightEndpoint: OntologyTypeVertexId;
+  }
+>;
+
+export const isLinksConstrainedByEdge = (
+  outwardEdge: OutwardEdge,
+): outwardEdge is LinksConstrainedByEdge => {
+  return outwardEdge.kind === "CONSTRAINS_LINKS_ON" && outwardEdge.reversed;
+};
+
+export type ConstrainsLinkDestinationsOnEdge = Subtype<
+  GenericOutwardEdge,
+  {
+    reversed: false;
+    kind: "CONSTRAINS_LINK_DESTINATIONS_ON";
+    rightEndpoint: OntologyTypeVertexId;
+  }
+>;
+
+export const isConstrainsLinkDestinationsOnEdge = (
+  outwardEdge: OutwardEdge,
+): outwardEdge is ConstrainsLinkDestinationsOnEdge => {
+  return (
+    outwardEdge.kind === "CONSTRAINS_LINK_DESTINATIONS_ON" &&
+    !outwardEdge.reversed
+  );
+};
+
+export type LinkDestinationsConstrainedByEdge = Subtype<
+  GenericOutwardEdge,
+  {
+    reversed: true;
+    kind: "CONSTRAINS_LINK_DESTINATIONS_ON";
+    rightEndpoint: OntologyTypeVertexId;
+  }
+>;
+
+export const isLinkDestinationsConstrainedByEdge = (
+  outwardEdge: OutwardEdge,
+): outwardEdge is LinkDestinationsConstrainedByEdge => {
+  return (
+    outwardEdge.kind === "CONSTRAINS_LINK_DESTINATIONS_ON" &&
+    outwardEdge.reversed
+  );
+};
+
+export type IsTypeOfEdge = Subtype<
+  GenericOutwardEdge,
+  {
+    reversed: true;
+    kind: "IS_OF_TYPE";
+    rightEndpoint: EntityValidInterval;
+  }
+>;
+
+export const isIsTypeOfEdge = (
```

Comment:
Yep, and I actually imported the wrong thing here :( which is why I was really thankful for my little tsc hacky tests below.

I've had some other ideas but they all come with other pitfalls which seem better but then end up being the same in my mind
- `isInstanceOfType`
- `hasType`

etc.

Reply:
`isTypeDescribing` / `isDescribedByType` ?

`isTypeConstraining` / `isConstrainedByType` ?

Reply:
I'd stay well clear of `constraining` because we use those for all of the ontology related edges (e.g. `constrainsPropertiesOn`), and I don't like `describing` :(